### PR TITLE
Added social share button in Blog page.

### LINKED ISSUE #14504: Istio blogs should have convenient buttons (for example X and linkedin button)
These are standards of blogs where each of them should have these convenient social site buttons to allow people to easily reshare.

saw it on cncf blog or cilium blog.

### PR DESCRIPTION
**Pull request by shivamgupta2020**
## Description

Added social share button for faster sharing of blog post on social sites.
Try to fix #14504 .

## Reviewers

- [ ] Ambient
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [X] User Experience
- [ ] Developer Infrastructure
- [ ] Localization/Translation

### CODE REVIEW

File: layouts/blog/single.html [hunk 1]
```diff
@@ -1,37 +1,69 @@
...
+        />
+      </a>
+      <a
+        href="https://twitter.com/intent/tweet?text={{ .Title }}&url={{ .Permalink }}"
+        target="_blank"
+      >
+        <img
+          src="https://img.icons8.com/ios/50/twitterx--v2.png"
+          style="width: 50px; height: 50px"
+          alt="twitterx--v2"
```
Comment (dhawton) [inline layouts/blog/single.html:52]:
```suggestion
          alt="Share to X"
```

File: layouts/blog/single.html [hunk 2]
```diff
@@ -1,37 +1,69 @@
...
+    <p class="social-share__title"><h4>SHARE</h4></p>
+    <div class="social-share__wrapper">
+      <a
+        href="https://www.linkedin.com/shareArticle?mini=true&url={{ .Permalink }}"
+        target="_blank"
+      >
+        <img
+          src="https://img.icons8.com/ios-glyphs/60/000000/linkedin.png"
+          style="width: 50px; height: 50px"
+          alt="linkedin"
```
Comment (dhawton) [inline layouts/blog/single.html:42]:
```suggestion
          alt="Share to LinkedIn"
```

File: layouts/blog/single.html [hunk 3]
```diff
@@ -1,37 +1,69 @@
...
+        />
+      </a>
+      <a
+        href="https://www.facebook.com/sharer/sharer.php?u={{ .Permalink }}"
+        target="_blank"
+      >
+        <img
+          src="https://img.icons8.com/ios-filled/50/000000/facebook-new.png"
+          style="width: 50px; height: 50px"
+          alt="facebook-new"
```
Comment (dhawton) [inline layouts/blog/single.html:62]:
```suggestion
          alt="Share to Facebook"
```

**Comment by kfaseela** [comment]
Please sign the CLA

**Review by dhawton — COMMENTED** [review]
Alts should be plain text, as they are used when images cannot be loaded and by some screen readers for accessibility. Also, I'd encourage getting SVGs of the icons and placing those in our static content dir instead of linking, or linking to a 1st party version, instead of to a 3rd party site.

File: layouts/blog/single.html [hunk 4]
```diff
@@ -1,37 +1,69 @@
...
+  </p>
+  {{ if .Page.Params.target_release }} {{ $page_parts := split
+  .Page.Params.target_release "." }} {{ $page_version := int (index $page_parts
+  0) }} {{ $page_revision := int (index $page_parts 1) }} {{
+  $page_normalized_version := add (add (mul $page_version 1000000) (mul
+  $page_revision 1000)) 0 }} {{ $site_parts := split .Site.Data.args.version "."
+  }} {{ $site_version := int (index $site_parts 0) }} {{ $site_revision := int
+  (index $site_parts 1) }} {{ $site_normalized_version := add (add (mul
+  $site_version 1000000) (mul $site_revision 1000)) 0 }} {{ if gt
+  $site_normalized_version $page_normalized_version }}
```
Comment (dhawton) [inline layouts/blog/single.html:19]: This formatting got changed here and is not readable. Can you remove the unnecessary changes like this?

**Comment by shivamgupta2020** [comment]
@dhawton I am confused on how the img are rendering in this site. I tried to use relative path but images doesn't load. 
On seeing other files I realized maybe "{{ partial "icon.html" "name" }}" may work, but I dont understand how exactly it will work. 
Can u suggest me where I can look or what I can do?

File: layouts/blog/single.html [hunk 5]
```diff
@@ -1,37 +1,69 @@
...
-    </article>
-{{ end }}
\ No newline at end of file
+    </aside>
+  </div>
+  {{ end }} {{ end }}
+  <div>{{ .Content }}</div>
+  <div class="social-share">
+    <p class="social-share__title"><h4>SHARE</h4></p>
+    <div class="social-share__wrapper">
```
Comment (dhawton) [inline layouts/blog/single.html:34]: You'll need to change the SASS as the class `.social-share` is set to `display: none !important` which hides this entire div. If you run `make serve` locally, it will run hugo which will display the site so you can try this out locally.

If you click "Details" next to the "test" `netlify/preliminary-istio/deploy-preview — Deploy Preview ready! ` you'll be taken to the live prelim view of this PR.

Comment (dhawton) [reply layouts/blog/single.html:34]: I don't see these other classes defined, so if they need to be they should be added otherwise remove them from here as they aren't needed.

**Comment by dhawton** [comment]
> @dhawton I am confused on how the img are rendering in this site. I tried to use relative path but images doesn't load. On seeing other files I realized maybe "{{ partial "icon.html" "name" }}" may work, but I dont understand how exactly it will work. Can u suggest me where I can look or what I can do?

You can place in /static/img and then from the image tag, `src="/img/..."`, for example. I'd personally create a /static/img/social/(site).svg and with img src="/img/social/(site).svg"

**Comment by shivamgupta2020** [comment]
@dhawton I have made changes regarding images and formatting. Please review.

File: layouts/blog/single.html [hunk 6]
```diff
@@ -1,37 +1,70 @@
...
       </div>
-      {{ partial "posts_navigation.html" . }}
-    </article>
-{{ end }}
\ No newline at end of file
+    </aside>
+  </div>
+  {{ end }} {{ end }}
+  <div>{{ .Content }}</div>
+    <p class="social-share__title"><h4>SHARE</h4></p>
```
Comment (dhawton) [inline layouts/blog/single.html:35]:
```suggestion
    <p><h4>Share this blog on social!</h4></p>
```

File: layouts/blog/single.html [hunk 7]
```diff
@@ -1,37 +1,70 @@
...
+    <div class="social-share__wrapper">
```
Comment (dhawton) [inline layouts/blog/single.html:36]:
```suggestion
    <div>
```

File: layouts/blog/single.html [hunk 8]
```diff
@@ -1,37 +1,70 @@
...
+      <a
+        href="https://www.linkedin.com/shareArticle?mini=true&url={{ .Permalink }}"
+        target="_blank"
+      >
+        <img
+          src="/img/social/linkedin.svg"
+          style="width: 50px; height: 50px"
```
Comment (dhawton) [inline layouts/blog/single.html:43]: Almost feel like this should be a class added to the blog SASS file instead of repeated here.

File: layouts/blog/single.html (hunk 7 above)
Comment (dhawton) [reply layouts/blog/single.html:36]: This class doesn't exist so serves no purpose here.

File: layouts/blog/single.html (hunk 6 above)
Comment (dhawton) [reply layouts/blog/single.html:35]: the style class `social-share__title` doesn't exist, so doesn't do anything here.

**Review by dhawton — COMMENTED** [review]
Some recommendations, the class ones were missed above.

File: src/sass/misc/_blog.scss [hunk 9]
```diff
@@ -216,6 +219,14 @@
         }
     }
 
+    .social-share__icon {
+        width: 50px;
+    }
+
+    img {
+        width: 50px;
+    }
```
Comment (dhawton) [inline src/sass/misc/_blog.scss]: I wouldn't do this as there may come a point where a post's images could be affected by this inadvertantly. The class you specified should be sufficient.

File: layouts/blog/single.html (hunk 6 above)
Comment (dhawton) [reply layouts/blog/single.html:35]: Can we fix the title here? `SHARE` looks out of place.

**Review by dhawton — APPROVED** [review]
LGTM. Thank you.
